packages/playground/website/src/lib/state/redux/boot-site-client.ts

```diff
@@ -0,0 +1,205 @@
+import { directoryHandleFromMountDevice } from '@wp-playground/storage';
+import { loadDirectoryHandle } from '../opfs/opfs-directory-handle-storage';
+import { getDirectoryPathForSlug } from '../opfs/opfs-site-storage';
+import {
+	addClientInfo,
+	removeClientInfo,
+	updateClientInfo,
+} from './slice-clients';
+import { logTrackingEvent } from '../../tracking';
+import { Blueprint, StepDefinition } from '@wp-playground/blueprints';
+import { logger } from '@php-wasm/logger';
+import { setupPostMessageRelay } from '@php-wasm/web';
+import { startPlaygroundWeb } from '@wp-playground/client';
+import { PlaygroundClient } from '@wp-playground/remote';
+import { getRemoteUrl } from '../../config';
+import { setActiveModal, setActiveSiteError } from './slice-ui';
+import { PlaygroundDispatch, PlaygroundReduxState } from './store';
+import { selectSiteBySlug } from './slice-sites';
+
+export function bootSiteClient(
+	siteSlug: string,
+	iframe: HTMLIFrameElement,
+	{ signal }: { signal: AbortSignal }
+) {
+	return async (
+		dispatch: PlaygroundDispatch,
+		getState: () => PlaygroundReduxState
+	) => {
+		signal.onabort = () => {
+			dispatch(removeClientInfo(siteSlug));
+		};
+		const site = selectSiteBySlug(getState(), siteSlug);
+
+		let mountDescriptor = undefined;
+		if (site.metadata.storage === 'opfs') {
+			mountDescriptor = {
+				device: {
+					type: 'opfs',
+					path: getDirectoryPathForSlug(site.slug),
+				},
+				mountpoint: '/wordpress',
+			} as const;
+		} else if (site.metadata.storage === 'local-fs') {
+			let localDirectoryHandle;
+			try {
+				localDirectoryHandle = await loadDirectoryHandle(site.slug);
+			} catch (e) {
+				logger.error(e);
+				dispatch(
+					setActiveSiteError(
+						'directory-handle-not-found-in-indexeddb'
+					)
+				);
+				return;
+			}
+			mountDescriptor = {
+				device: {
+					type: 'local-fs',
+					handle: localDirectoryHandle,
+				},
+				mountpoint: '/wordpress',
+			} as const;
+		}
+
+		let isWordPressInstalled = false;
+		if (mountDescriptor) {
+			try {
+				isWordPressInstalled = await playgroundAvailableInOpfs(
+					await directoryHandleFromMountDevice(mountDescriptor.device)
+				);
+			} catch (e) {
+				logger.error(e);
+				if (e instanceof DOMException && e.name === 'NotFoundError') {
+					dispatch(
+						setActiveSiteError(
+							'directory-handle-not-found-in-indexeddb'
+						)
+					);
+					return;
+				}
+				dispatch(setActiveSiteError('directory-handle-unknown-error'));
+				return;
+			}
+		}
+
+		logTrackingEvent('load');
+
+		let blueprint: Blueprint;
+		if (isWordPressInstalled) {
+			blueprint = site.metadata.runtimeConfiguration;
+		} else {
+			blueprint = site.metadata.originalBlueprint;
```

Comment:
Nice.

Reply:
Thanks! We might need to rethink this to support [multiple Blueprints](https://github.com/WordPress/blueprints-library/issues/49#issuecomment-2353756396), but that's way out of scope for this PR.